### PR TITLE
Feature/probes

### PR DESCRIPTION
**Pull request by mrapacz**

### CODE REVIEW

File: nodemcu/temperature.lua [hunk 1]
```diff
@@ -7,8 +7,11 @@ end
 
 function module.get_temperature()
     
-    temp = t.read() --used to trigger measuring
-    return t.read()
+    temp = t.read()
+    while(temp == 85) do
```
Comment (klausweiss) [inline nodemcu/temperature.lua]: I believe that magic numbers should be in constants.

File: raspberry/weatherserver/temperatures/models.py [hunk 2]
```diff
@@ -7,3 +7,6 @@ class Probe(models.Model):
     node_id = models.IntegerField()
     temperature = models.FloatField()
     timestamp = models.DateTimeField(default=timezone.now)
+
+    def __str__(self):
+        return " ".join((str(x) for x in [self.temperature, self.timestamp, self.node_id, ]))
```
Comment (klausweiss) [inline raspberry/weatherserver/temperatures/models.py]: i find `"{} {} {}".format(self…)` more readable if there are just three arguments.

File: raspberry/weatherserver/temperatures/templates/temperatures/index.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+<body>
+
+<h1> Temperatures</h1>
+
+{% block content %}
+    <div id="container">Chart</div>
+{% endblock %}
+<h2>Last update:</h2>
+<h3>{{ last_measured }} measured at {{ last_date }}</h3>
+{#<ul>#}
```
Comment (klausweiss) [inline raspberry/weatherserver/temperatures/templates/temperatures/index.html]: You should get rid of commented code.

File: raspberry/weatherserver/weatherserver/settings.py [hunk 4]
```diff
@@ -38,8 +41,10 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-    # 'temperatures' if __name__ == 'weatherserver.settings' else 'weatherserver.temperatures'
-    'temperatures'
+    # 'jquery',  # for chartit app
```
Comment (klausweiss) [inline raspberry/weatherserver/weatherserver/settings.py]: You should get rid of commented code.

File: raspberry/weatherserver/weatherserver/settings.py [hunk 5]
```diff
@@ -109,7 +114,7 @@
 
 LANGUAGE_CODE = 'en-us'
 
-TIME_ZONE = 'CET'
+TIME_ZONE = 'Europe/Warsaw'
```
Comment (klausweiss) [inline raspberry/weatherserver/weatherserver/settings.py:115]: Do you really want timezone to be hardcoded?

File: raspberry/weatherserver/temperatures/views.py [hunk 6]
```diff
@@ -1,5 +1,57 @@
...
+            'xAxis': {'title': {'text': 'Time'}},
+            # -*- coding: utf-8 -*-
+            'yAxis': {'title': {'text': 'Temperature (°C)'}}
+
+        },
+        x_sortf_mapf_mts=(None, tz_oriented_date, False)
+    )
+    # -*- coding: utf-8 -*-
+    last_probe = Probe.objects.order_by('-timestamp')[0]
+    last_measured = str(last_probe.temperature) + "°C"
```
Comment (klausweiss) [inline raspberry/weatherserver/temperatures/views.py]: I find `"{}°C".format(last_probe.temperature)` more readable.

File: raspberry/weatherserver/temperatures/templates/temperatures/index.html [hunk 7]
```diff
@@ -0,0 +1,36 @@
+<head>
```
Comment (klausweiss) [inline raspberry/weatherserver/temperatures/templates/temperatures/index.html]: Where are `doctype` and `html`?

File: raspberry/weatherserver/temperatures/templates/temperatures/index.html [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        line-height: 1.2
+    }</style>
+    <script src="https://code.jquery.com/jquery-1.12.4.min.js"></script>
+    <script src="https://code.highcharts.com/highcharts.js"></script>
+    {% load chartit %}
+    {{ chart|load_charts:"container" }}
+</head>
+<body>
+
+<h1> Temperatures</h1>
```
Comment (klausweiss) [inline raspberry/weatherserver/temperatures/templates/temperatures/index.html]: I would indent that :P

File: raspberry/weatherserver/temperatures/models.py (hunk 2 above)
Comment (mrapacz) [reply raspberry/weatherserver/temperatures/models.py]: aye, much better

File: nodemcu/temperature.lua (hunk 1 above)
Comment (mrapacz) [reply nodemcu/temperature.lua]: I agree

File: raspberry/weatherserver/weatherserver/settings.py (hunk 5 above)
Comment (mrapacz) [reply raspberry/weatherserver/weatherserver/settings.py:115]: Yep.

**Review by klausweiss — APPROVED** [review]

**Pull request merged**